program/rust/src/tests/test_sma.rs

```diff
@@ -0,0 +1,168 @@
+use quickcheck::Arbitrary;
+use quickcheck_macros::quickcheck;
+
+use crate::time_machine_types::{
+    DataPoint,
+    SmaTracker,
+    NUM_BUCKETS_THIRTY_MIN,
+};
+
+#[derive(Clone, Debug, Copy)]
+struct DataEvent {
+    timegap:  i64,
+    slot_gap: u64,
+    price:    i64,
+}
+
+impl Arbitrary for DataEvent {
+    fn arbitrary(g: &mut quickcheck::Gen) -> Self {
+        DataEvent {
+            timegap:  i64::from(u8::arbitrary(g)),
+            slot_gap: u64::from(u8::arbitrary(g)) + 1, /* Slot gap is always > 1, because there
+                                                        * has been a succesful aggregation */
+            price:    i64::arbitrary(g),
+        }
+    }
+}
+
+
+#[quickcheck]
+fn test_sma(input: Vec<DataEvent>) -> bool {
+    // No gaps, no skipped epochs
```

Comment:
suggest leaving a comment above this explaining how the tests are set up, specifically that the timegap / slot_gap can be at most `u8::MAX`. It took me a bit to understand these comments about skips / gaps.